docs/azure_authentication.md

```diff
@@ -0,0 +1,52 @@
+# Azure Operator authentication with Azure
+Both the Azure operator and Kubernetes need to send authenticated requests to the Azure API. That means we have different processes talking to the Azure API in different ways
+
+- Azure operator running on the control plane worker nodes
+- `credentiald` running on the control plane worker nodes
+- Kubernetes cloud provider integration running on the control plane master nodes
+- Kubernetes cloud provider integration running on the control plane worker nodes
+- Kubernetes cloud provider integration running on tenant clusters master nodes
+- Kubernetes cloud provider integration running on tenant clusters worker nodes
+			
+## Azure Operator
+This process runs on the worker nodes on the control planes. [When deployed](https://github.com/giantswarm/installations/blob/ad518ba95584685bf37cd9c3cca403115153dc14/ghost/draughtsman-secret-values.yaml#L20-L27), it receives a set of [parameters](https://github.com/giantswarm/azure-operator/blob/d0185b1de2cf6975f8e795a38bb55dfc875ecdeb/main.go#L125-L126) 
```

Comment:
Correct me if I'm wrong, but I think what I wrote is not 100% true. 

The Azure Operator only uses the credentials parameters to set up some DNS and VPN resources. It uses a different set of credentials to create the clusters: the credentials stored in a Kubernetes secret referenced from the `AzureConfig` object. 
These are typically the same set of credentials, but they don't have to be.

Can you confirm my understanding is correct? In that case, I'd add that to the document.

Reply:
Yes it uses configured credentials to set DNS and VPN gateways in CP subscription and then it uses so called BYOC (bring your own cloud) credentials referenced in AzureConfig CR. They are configured per cluster basis. This is because we allow to create clusters in different subscriptions. This is similar for AWS.

Reply:
Updated